### ISSUE TITLE
Create event detail page (create/edit)

### ISSUE THREAD

**Issue by blakestone95** [blakestone95/AshkeepersRestWebsite#67]
Description:
Users need a page where they can an event.  Similarly, users also need the ability to edit existing events. They should be able to edit the event's title, start/end dates, and description.  In order to facilitate creativity, users ought to be able to use markdown to nicely format the descriptions they write.

No design yet.

Acceptance Criteria:
- Only users with editor and admin permissions can create a new event
- Only an admin or the event's original creator can edit or delete an event
- The following are editable fields
  - Title
  - Start date
  - End date
  - Description
- The description field has a markdown editor which allows for previewing the content (might make this a separate issue)
- There are action buttons at the bottom of the page
  - For new events
    - Create - saves the new event and redirects to that new event's detail view
    - Cancel - returns user to events list
  - For existing event
    - Edit - visible initially when viewing; puts the event detail page into edit mode
    - Cancel - visible when editing; ignores changes and returns the page to view mode
    - Save - visible when editing; saves the changes and returns the page to view mode with the new edits reflected
    - Delete - visible at all times; deletes the event from the database entirely
      - Deletion cannot occur unless the user confirms the action in a confirmation modal

References to other Issues:
Split and clarified from #11
Blocked by #65, #66, #?? (<- user permissions issue yet to be created)
